rfcs/20200612-stream-executor-c-api.md

```diff
@@ -336,94 +401,145 @@ typedef struct SP_StreamExecutor {
                            TF_Status* status);
 
   // Causes the host code to synchronously wait for the event to complete.
-  void (*block_host_for_event)(
-      SP_Device* executor, SP_Event event, TF_Status* status);
+  void (*block_host_for_event)(const SP_Device* device, SP_Event event,
+                               TF_Status* status);
+
+  // [Optional]
+  // Causes the host code to synchronously wait for operations entrained onto
+  // stream to complete. Effectively a join on the asynchronous device
+  // operations enqueued on the stream before this program point.
+  // If not set, then corresponding functionality will be implemented
+  // by registering an event on the `stream` and waiting for it using
+  // `block_host_for_event`.
+  void (*block_host_until_done)(const SP_Device* device, SP_Stream stream,
+                                TF_Status* status);
 
   // Synchronizes all activity occurring in the StreamExecutor's context (most
   // likely a whole device).
-  TF_BOOL (*synchronize_all_activity)(SP_Device* executor);
-
-  // Obtains metadata about the underlying device.
-  void (*fill_device_description)(SP_Device* executor,
-                                  SP_DeviceDescription* description,
-                                  TF_Status* status);
+  void (*synchronize_all_activity)(const SP_Device* device, TF_Status* status);
 
   // Enqueues on a stream a user-specified function to be run on the host.
-  TF_BOOL (*host_callback)(SP_Device* executor, SP_Stream* stream,
-                           TF_StatusCallbackFn callback_fn, void* ctx);
+  // `callback_arg` should be passed as the first argument to `callback_fn`.
```

Comment:
```suggestion
  // `callback_arg` will be passed as the first argument to `callback_fn`.
```

Reply:
This is an instruction for the plugin author. Therefore "should" or "must" seems better.

Reply:
It needs to be passed into callback_fn as part of the design requirement. Stating `should` leaves room for ambiguity but does seem more polite. How about _shall_?

Reply:
I'm okay with either. "Shall" does sound a bit more forceful than "should" and more polite than "must". Let me know what you agree upon. :)

Reply:
All sound good to me. As a non-native speaker, I can't really tell the small differences between the three :)
I just though `will` sounds like the argument will be passed by itself. However, it is plugin responsibility to make sure it is passed in.

Reply:
Oh, I didn't see that it was "will" before. Alright, "shall" shall be used then.